### PR TITLE
fix(nuxt,kit): getNuxtModuleVersion not returning version

### PR DESCRIPTION
**Pull request by TonyFresneau**
### 📚 Description

It all started with a warning when using the @nuxtjs/sitemap module.

image

I ended up testing the `getNuxtModuleVersion` method, but it wasn't returning the version of my module.

So, I added `nuxtModule.meta.version` in the return condition. In my case, this solved the issue, as I’m using a remote monorepo composed of layers, one of which includes `@nuxtjs/i18n`. This solution might also help in other similar configurations.

## Summary by CodeRabbit

- **New Features**
	- Improved version retrieval for Nuxt modules, ensuring more reliable access to module metadata.

- **Bug Fixes**
	- Enhanced error handling in the version retrieval process, providing fallback options when build-time metadata is unavailable.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]
good idea!

File: packages/kit/src/module/compatibility.ts [hunk 1]
```diff
@@ -58,8 +58,8 @@ export async function getNuxtModuleVersion (module: string | NuxtModule, nuxt: N
   }
   // it's possible that the module will be installed, it just hasn't been done yet, preemptively load the instance
   if (hasNuxtModule(moduleMeta.name)) {
-    const { buildTimeModuleMeta } = await loadNuxtModuleInstance(moduleMeta.name, nuxt)
-    return buildTimeModuleMeta.version || false
+    const { nuxtModule, buildTimeModuleMeta } = await loadNuxtModuleInstance(moduleMeta.name, nuxt)
+    return buildTimeModuleMeta.version || nuxtModule.meta.version || false
```
Comment (danielroe) [inline packages/kit/src/module/compatibility.ts:62]:
```suggestion
    return buildTimeModuleMeta.version || await nuxtModule.getMeta?.().then(r => r.version) || false
```

**Review by danielroe — COMMENTED** [review]
does my change still resolve things for you @Aurion72?

**Comment by Aurion72** [comment]
@danielroe Hello, yes, it continues to work with your changes ;)

**Review by danielroe — APPROVED** [review]

**Comment by cyrilf** [comment]
Hey @Aurion72 , I had a similar issue. Any chance your fix is going to fix this one too: https://github.com/nuxt-modules/sitemap/issues/379 ? Thanks!

**Comment by Aurion72** [comment]
Hello @cyrilf yes, it will fix that too i suppose. @danielroe merge my PR to main, we need to wait a new release by now :)

Have a good day !

**Pull request merged**